starknet_py/net/udc_deployer/deployer.py

```diff
@@ -0,0 +1,146 @@
+from starknet_py.utils.data_transformer.universal_deployer_serializer import (
+    universal_deployer_serializer,
+    deploy_contract_abi,
+    deploy_contract_event_abi,
+)
+
+
+class Deployer:
+    """
+    Deployer used to deploy contracts through Universal Deployer Contract (UDC)
+    """
+
+    def __init__(
+        self,
+        account: AccountClient,
+        address: Optional[AddressRepresentation] = None,
+        salt: Optional[int] = None,
+        unique: bool = True,
+    ):
+        """
+        :param account: AccountClient used to sign and send transactions
+        :param address: Address of the UDC. Must be set when using a custom network
+        :param salt: The salt for a contract to be deployed. Random value is selected if it is not provided
+        :param unique: Boolean determining if the salt should be connected with the account's address. Default to True
+        """
+        address = (
+            address if isinstance(address, int) or address is None else int(address, 16)
+        )
+        address = address or deployer_address_from_network(
+            net=account.net, deployer_address=address
+        )
+
+        self.account = account
+        self.address = address
+        self.salt = salt
+        self.unique = unique
+
+    def make_deployment(self, class_hash: Hash, abi: Optional[List] = None):
+        """
+        Creates a ContractDeployment instance which is used to prepare and send deploy invoke transactions
+
+        :param class_hash: The class_hash of the contract to be deployed
+        :param abi: ABI of the contract to be deployed
+        :returns: ContractDeployment
+        """
+
+        return Deployer.ContractDeployment(
+            deployer=self, class_hash=class_hash, abi=abi
+        )
+
+    async def get_deployed_contract_address(self, transaction_hash: Hash) -> int:
+        """
+        Returns deployed contract address
+
+        :param transaction_hash: Hash of the already send and accepted deploy invoke transaction
+        :returns: An address of the deployed contract
+        """
+        event = await self._get_deploy_event(transaction_hash=transaction_hash)
+
+        if not event:
+            raise ValueError(
+                "ContractDeployed event was not found."
+                "Make sure that transaction_hash is the hash of UDC deploy transaction"
+            )
+
+        event = universal_deployer_serializer.to_python(
+            value_types=deploy_contract_event_abi["data"],
+            values=event.data,
+        )
+
+        return event[0]
+
+    async def _get_deploy_event(self, transaction_hash: Hash) -> Optional[Event]:
+        receipt = await self.account.get_transaction_receipt(tx_hash=transaction_hash)
+
+        for event in receipt.events:
+            if get_selector_from_name("ContractDeployed") == event.keys[0]:
+                return event
+        return None
```

Comment:
```suggestion
next(filter(lambda event: get_selector_from_name("ContractDeployed") == event.keys[0], receipt.events), None)
```
I'm not sure if this is simpler than the for loop but certainly is 1 line vs 4 😅

Reply:
It is very dense and harder to read.